### PR TITLE
Minor copyright/v2

### PR DESCRIPTION
**Pull request by inashivb**
Previous PR: https://github.com/OISF/suricata/pull/7795

Changes since last PR:
- Fix copyrights for non range years
- Make updates to the entire codebase
- clang fmt changes done separately as they seemed intrusive

### CODE REVIEW

File: rust/cbindgen.toml [hunk 1]
```diff
@@ -10,7 +10,7 @@ language = "C"
 
 # An optional string of text to output at the beginning of the generated file
 # default: doesn't emit anything
-header = """/* Copyright (C) 2019 Open Information Security Foundation
+header = """/* Copyright (C) 2019-2022 Open Information Security Foundation
```
Comment (inashivb) [inline rust/cbindgen.toml:13]: Should probably be 2022?

**Comment by inashivb** [comment]
@jufajardini could you please help review this?

File: rust/cbindgen.toml (hunk 1 above)
Comment (jufajardini) [reply rust/cbindgen.toml:13]: Makes sense to me!

File: rust/src/detect.rs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/* Copyright (C) 2022 Open Information Security Foundation
+/* Copyright (C) 2022-2022 Open Information Security Foundation
```
Comment (jufajardini) [inline rust/src/detect.rs:1]: Another exception case! :P

**Review by jufajardini — CHANGES_REQUESTED** [review]
Found some files that were created on 2022 and which won't need the copyright update either, commented on one of them :)

**Comment by catenacyber** [comment]
Did you use a script to create this and modify > 1k files ?
Can it be run in CI ?

**Comment by inashivb** [comment]
> Did you use a script to create this and modify > 1k files ? 

Yeah. Kind of. I used a grep + sed oneliner.

> Can it be run in CI ?

I was thinking that maybe we could do this the start of every year. Now that all the files are current, what should we add in the CI?
Or do you mean using CI as an indicator to do this kind of an update. That's actually much more reliable than remembering to do this every year indeed.
Did I get you right?

File: rust/src/detect.rs (hunk 2 above)
Comment (inashivb) [reply rust/src/detect.rs:1]: Thanks. :)
Fixed in the new rev.

**Comment by catenacyber** [comment]
> > Did you use a script to create this and modify > 1k files ?
> 
> Yeah. Kind of. I used a grep + sed oneliner. But, there are some corner cases.
> 

Can you put this one-liner in the commit description ?

> > Can it be run in CI ?
> 
> I was thinking that maybe we could do this the start of every year. Now that all the files are current, what should we add in the CI? Or do you mean using CI as an indicator to do this kind of an update. That's actually much more reliable than remembering to do this every year indeed. Did I get you right?

Should every file have an updated copyright to 2022 ? Or just until the last year it was modified ?

**Comment by inashivb** [comment]
> Can you put this one-liner in the commit description ?

Yes. I'll write down a script for the series of commands. 
 
> Should every file have an updated copyright to 2022 ? Or just until the last year it was modified ?

hmmm i thought it was always supposed to be current. i am not very sure. No one seemed to mention this either.

**Comment by jufajardini** [comment]
> > Can you put this one-liner in the commit description ?
> 
> Yes. I'll write down a script for the series of commands.
> 
> > Should every file have an updated copyright to 2022 ? Or just until the last year it was modified ?
> 
> hmmm i thought it was always supposed to be current. i am not very sure. No one seemed to mention this either.

I had a feeling that we did the "update-when-change" thing because it was more practical, only...

**Comment by inashivb** [comment]
> I had a feeling that we did the "update-when-change" thing because it was more practical, only...

In that case, let's take other folks' opinion in. Maybe this is not needed at all then.

**Pull request closed without merging**